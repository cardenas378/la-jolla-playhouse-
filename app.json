{
  "addons": [

  ],
  "buildpacks": [

  ],
  "env": {
  },
  "formation": {
  },
  "name": "la-jolla-playhouse",
  "scripts": {
  },
  "stack": "heroku-18"
}
