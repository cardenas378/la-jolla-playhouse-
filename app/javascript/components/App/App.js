import PropTypes from "prop-types";
import React from "react";
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Link
} from "react-router-dom";

import styled, { ThemeProvider } from "styled-components/macro";

import { Home } from "../../Pages/Home";
import { Locations } from "../../Pages/Locations";
import { AddBooking } from "../AddBooking";
import { AddLocation } from "../AddLocation";
import { getLocations, createLocation } from "../api";
import { NavBar } from "../NavBar";
import GlobalStyle from "../styles/Global";
import Theme from "../styles/Theme";

import { StyledLayout } from "./App.styles";

import {
  Header,
  AddBookingHeader,
  FormSectionHeader,
  StyledSelect,
  FullInput,
  RadioButton,
  SubmitButton,
  SaveBookingButton,
  HalfInput,
  ThirdInput
} from "..";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      locations: [],
      success: false
    };
  }

  componentDidMount() {
    getLocations().then(locations => {
      this.setState({ locations });
    });
  }

  handleNewLocation = location => {
    createLocation(location).then(location => {
      this.setState({ success: true });
    });
  };

  changeSuccess = value => {
    this.setState({ success: value });
  };

  componentDidMount() {
    getLocations().then(locations => {
      this.setState({ locations });
    });
  }

  render() {
    const { locations, success } = this.state;

    return (
      <ThemeProvider theme={Theme}>
        <GlobalStyle />
        <StyledLayout>
          <Router>
            <NavBar />

            <main>
              <Route
                exact
                path="/AddLocation"
                render={props => {
                  return (
                    <AddLocation
                      {...props}
                      handleNewLocation={this.handleNewLocation}
                      success={success}
                    />
                  );
                }}
              />

              <Route
                exact
                path="/Locations"
                render={props => {
                  return (
                    <Locations
                      {...props}
                      changeSuccess={this.changeSuccess}
                      locations={locations}
                      success={success}
                    />
                  );
                }}
              />

              <Route
                exact
                path="/"
                render={props => {
                  return <Home {...props} />;
                }}
              />
              <Route
                exact
                path="/AddBooking/:location_id"
                render={props => {
                  return (
                    <AddBooking
                      {...props}
                      changeSuccess={this.changeSuccess}
                      locations={this.state.locations}
                      success={this.state.success}
                    />
                  );
                }}
              />
            </main>
          </Router>
        </StyledLayout>
      </ThemeProvider>
    );
  }
}

App.propTypes = {};

export { App };
