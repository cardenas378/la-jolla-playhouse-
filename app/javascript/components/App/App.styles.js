import styled from "styled-components";

const StyledLayout = styled.div`
  display: grid;
  grid-template-columns: 225px 1fr;
  grid-template-areas: "sidebar main";

  main {
    grid-area: main;
    background: ${props => props.theme.whiteSmoke};
    min-height: 100vh;
  }
`;

export { StyledLayout };
