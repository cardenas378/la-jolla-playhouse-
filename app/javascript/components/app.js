import React from "react";
import styled, { ThemeProvider } from 'styled-components/macro';
import { BrowserRouter as Router, Route, Link } from "react-router-dom"
import { getLocations, createLocation, createBooking } from './api'
import AddLocation from './AddLocation'
import AddBooking from './AddBooking'
import Home from './Home'
import Theme from './styles/Theme'
import GlobalStyle from './styles/Global'
import Logo from "./Logo";

const StyledLayout = styled.div`
  display: grid;
  grid-template-columns: 225px 1fr;
  grid-template-areas:
    "sidebar main";

  aside {
    border-right: 1px solid ${props => props.theme.whiteSmoke};
    display: flex;
    flex-direction: column;
    height: 100vh;
    grid-area: sidebar;
    padding: 45px;
    position: fixed;

     ol {
       list-style-type: none;
       padding: 0;
     }

    ol.top-nav {
      counter-reset: nav-list;
      margin: 32px 0 0 0;

      li:before {
        color: ${props => props.theme.whiteSmoke};
        content: counter(nav-list, decimal-leading-zero);
        counter-increment: nav-list;
        font-family: ${props => props.theme.serif};
        left: -25px;
        position: absolute;
      }
    }

    ol.btm-nav {
      justify-self: flex-end;
      margin: auto 0 0 0;
    }

    li {
      font-family: ${props => props.theme.serif};
      font-size: 1.4rem;
      letter-spacing: 1px;
      margin-bottom: 15px;
      position: relative;
      text-transform: uppercase;

      a {
        color: ${props => props.theme.zircon};
        font-weight: normal;
        text-decoration: none;
      }
    }
  }

  main {
    grid-area: main;
  }
`;

class App extends React.Component{
  constructor(props) {
    super(props)
    this.state = {
      locations: [],
      bookings: [],
      success: false
    }
  }

  handleNewLocation = (location) => {
       createLocation(location)
       .then(location => {
         this.setState({success: true})
         getLocations()
         .then(locations=>{
             this.setState({locations})
         })
     })
   }

  handleNewBooking = (booking) => {
       createBooking(booking)
       .then(booking => {
         this.setState({success: true})
         getBookings()
         .then(bookings=>{
             this.setState({bookings})
         })
     })
   }

  changeSuccess = (value) => {
    this.setState({success: value})
  }

  componentDidMount() {
      getLocations()
      .then( locations => {
            this.setState({locations})
      })
  }

  // componentDidUpdate(prevState) {
  //     if(prevState.locations != this.state.locations) {
  //         this.setState({success: false})
  //     }
  // }

  render() {
    let {locations} = this.state
  return(
    <ThemeProvider theme={Theme}>
      <GlobalStyle />
      <StyledLayout>
    <Router>
      <aside>
        <Logo />
        <ol className="top-nav">
            <li><Link to="/">Home</Link></li>
            <li><Link to="/AddLocation">Add Location</Link></li>
            <li><Link to="/AddBooking">Add Booking</Link></li>
        </ol>
        <ol className="btm-nav">
          <li><Link>Settings</Link></li>
          <li><Link>Logout</Link></li>
        </ol>
      </aside>

      <main>
      <Route exact path="/AddLocation" render={(props)=>{
        return(
          <AddLocation
              {...props} handleNewLocation={this.handleNewLocation} success={this.state.success}
          />
        )
      }}
      />
      <Route exact path="/AddBooking" render={(props)=>{
        return(
          <AddBooking
              {...props} handleNewBooking={this.handleNewBooking} success={this.state.success} locations={this.state.locations}
          />
        )
      }}
      />
      <Route exact path="/" render={(props)=>{
        return(
          <Home
              {...props} success={this.state.success} changeSuccess={this.changeSuccess} locations={this.state.locations}
          />
        )
      }}
      />
      </main>
    </Router>
    </StyledLayout>
    </ThemeProvider>

  )
}
}
export default App;
