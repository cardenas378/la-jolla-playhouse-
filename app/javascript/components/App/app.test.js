import { shallow } from "enzyme";
import React from "react";
import ReactDOM from "react-dom";
import Enzyme from "enzyme"
import Adapter from 'enzyme-adapter-react-16'

<<<<<<< HEAD
import { App } from "../app";
=======
import { App } from "./App";

Enzyme.configure({ adapter: new Adapter() });
jest.mock("../api/");
>>>>>>> 5574cefe17b005d8521a492a52597c6b056b121a

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<App />, div);
});

describe("App", () => {
  test("renders the component correctly", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
