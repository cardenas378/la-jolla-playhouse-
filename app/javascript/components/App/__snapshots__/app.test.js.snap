// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "alabaster": "#fef217",
      "black": "#212121",
      "broom": "#fef217",
      "charcoal": "#424242",
      "contentPaddingLeft": "65px",
      "crayonOrange": "#f26726",
      "doveGrey": "#757575",
      "error": "#b92121",
      "frenchGray": "#bdbdbd",
      "funBlue": "#1862a6",
      "georgia": "georgia, serif",
      "melon": "#fbbfb7",
      "nunito": "Nunito Sans, sansSerif",
      "offsetContentPaddingLeft": "-65px",
      "persianIndigo": "#66239c",
      "pureWhite": "#ffffff",
      "sansSerif": "Nunito Sans, sansSerif",
      "serif": "georgia, serif",
      "starDust": "#9e9e9e",
      "stormDust": "#616161",
      "success": "#29a028",
      "warning": "#f7df5a",
      "weightBlack": 900,
      "weightBold": 700,
      "weightExtraLight": 200,
      "weightLight": 300,
      "weightRegular": 400,
      "white": "#eeeeee",
      "whiteSmoke": "#f5f5f5",
      "zircon": "#e0e0e0",
    }
  }
>
  <GlobalStyleComponent />
  <styled.div>
    <BrowserRouter>
      <NavBar />
      <main>
        <Route
          exact={true}
          path="/AddLocation"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/AddBooking"
          render={[Function]}
        />
      </main>
    </BrowserRouter>
  </styled.div>
</ThemeProvider>
`;
